lazy.go

```diff
@@ -51,7 +51,11 @@ type lazyConn struct {
 }
 
 func (l *lazyConn) Read(b []byte) (int, error) {
-	if !l.rhandshake {
+	l.rhlock.RLock()
+	rhandshake := l.rhandshake
+	l.rhlock.RUnlock()
```

Comment:
Ok, we can probably use `internals/race`. However, we can probably get a way with using `sync.Once` without losing much, if any, performance.